packages/gatsby-plugin-graphql-codegen/src/gatsby-node.ts

```diff
@@ -75,10 +158,10 @@ export const onPostBootstrap: NonNullable<GatsbyNode['onPostBootstrap']> = async
       return
     }
     const { schema } = store.getState()
-    await buildDebounce(schema)
+    await buildDebounce(await prepareSchemas(schema, additionalSchemas))
```

Comment:
does it make sense to prepare the schemas once before the build? I'm somewhat concerned about this line because of the await before the debounce. If there is a potential longer async task, I think it makes sense to debounce it.

so the way I see it, we have two options:

1. debounce `prepareSchemas` with `build`
2. prepare the schemas once before build (preferred & simpler)

What do you think?

cc @d4rekanguok

Reply:
You're absolutely right, totally missed that, the `prepareSchema` call should be debounced as well in this case.

The idea here was to update the schemas before each codegen, so that if the schema updates, those changes are "instantaneously" reflected in the generated types (as it happens in the currently released version of the plugin if the gatsby schema changed during "runtime"). However, I think this is rarely the case and it complicates the implementation to do this right, so if you agree, I would change the implementation to only prepare the schemas once before build (option 2).

If at a later date it becomes clear that updating the schemas continuously is a requirement, it could be implemented as a new feature. Maybe then it could be implemented so that the schemas are updated in a fixed interval (like `gatsby-source-graphql` does with the `refetchInterval` option) instead of before every codegen.

Reply:
Yeah this is an interesting product question. I prefer option 2 too, I think it's simpler to stick to generating it once at startup but I can see it being annoying to have to restart the gatsby server if you change the schema but my bet is users don't dev both the client and server at the same time.

I'll let @d4rekanguok have the final word on that

Reply:
Yeah lets go with option 2!

Reply:
Ok, great. I've just updated the implementation :+1: